src/main/java/bio/terra/service/snapshotbuilder/SnapshotRequestDao.java

```diff
@@ -103,6 +105,24 @@ public List<SnapshotAccessRequestResponse> enumerateByDatasetId(UUID datasetId)
     }
   }
 
+  /**
+   * Return the list of Snapshot Requests associated with the given snapshot id.
+   *
+   * @param snapshotId associated with any number of snapshot requests.
+   * @return the list of snapshot requests, empty if none, or an exception if the snapshot does not
+   *     exist.
+   */
+  @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+  public List<SnapshotAccessRequestResponse> enumerateBySnapshotId(UUID snapshotId) {
+    String sql = "SELECT * FROM snapshot_request WHERE snapshot_id = :snapshot_id";
+    MapSqlParameterSource params = new MapSqlParameterSource().addValue(SNAPSHOT_ID, snapshotId);
+    try {
+      return jdbcTemplate.query(sql, params, responseMapper);
+    } catch (EmptyResultDataAccessException ex) {
+      throw new NotFoundException("No snapshot requests found for given snapshot id", ex);
```

Comment:
If the error only pops up when there's no snapshot, this error message is wrong, I think. Should it be `No snapshot requests found for...` or `No snapshot found for ...`

Reply:
I think this can occur in either case. If the user passes an invalid snapshot ID, I think a `404` is still correct although I agree that the error, while still technically correct, may lead the user to think that the snapshot ID was valid. I don't think it's worth adding a separate check for a valid snapshot ID since a user can do that themselves, so instead I'd make the message more vague.
```suggestion
      throw new NotFoundException("No snapshot requests found for given snapshot id, or the id isn't a valid snapshot id", ex);
```